validatePassword.js

```diff
@@ -0,0 +1,20 @@
+const validatePassowrd = (password) => {
+  const specialCharacterCheck = /!/
+  const numberCheck = /\d/
+  const upperCaseCheck = /[A-Z]/
+  const lowerCaseCheck = /[a-z]/
+
+  if (specialCharacterCheck.test(password)) {
+    if (numberCheck.test(password)) {
+      if (upperCaseCheck.test(password)) {
+        if (lowerCaseCheck.test(password)) {
+          if (password.length >= 8) {
+            return true
+          }
+        }
+      }
+    }
+  }
```

Comment:
This is what we in dev call "the pyramid of doom" and it is definitely and anti-pattern. Plus it isn't necessary since you are saying if all of these things are true return true. That's what `&&` is for.

```JavaScript
const validatePassowrd = (password) => {
  const specialCharacterCheck = /!/
  const numberCheck = /\d/
  const upperCaseCheck = /[A-Z]/
  const lowerCaseCheck = /[a-z]/

  return specialCharacterCheck.test(password)) &&
    numberCheck.test(password) &&
    upperCaseCheck.test(password) &&
    lowerCaseCheck.test(password) &&
    password.length >= 8
}
```